README.md

```diff
@@ -113,6 +113,125 @@ function ApiRequestHandler(request, reply) {
 
 In the above example *isolated_db* is the same instance of TinyPg except with an overridden events property and *dispose* method to remove all listeners. The *UserService* can create other services and pass its reference to *isolated_db* to other services. In doing so, you can track all database queries executed as the result of every API request.
 
+## Hooks
```

Comment:
I'm not sure an error should result in a noop.

Reply:
The thought there is that consistent hook errors would render that instance of tiny unusable.

Also depending on which hook throws the error (`onResult` for example) the query might have already been submitted which would be confusing to the user.

Reply:
Could take a note from [React component error handling](https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html#introducing-error-boundaries) and add an error property to the context or something

Reply:
If the purpose is to transform the input to the database you wouldn’t want a query to execute. 

I can see if you’re simply looking into the lifecycle for monitoring. But the transformation seems to make it different.

Reply:
Ah ok sure, `preSql` and `preRawQuery` (the only transformative hooks) should throw but the others should not. I like that idea